### PR TITLE
Migrate to CircleCI 2.0

### PR DESCRIPTION
**Pull request by abesto**
As part of https://github.com/openzipkin/zipkin/issues/2102; based heavily on previous work in https://github.com/openzipkin/brave/pull/725 and https://github.com/openzipkin/zipkin-api/pull/52

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+      - run:
+          name: Tests
+          command: ./mvnw clean verify
+
+      - run:
+          name: Collect test reports
+          command: |
+            mkdir -p /tmp/test-reports/unit-tests/
+            find . -type f -regex ".*/target/surefire-reports/.*xml" -exec cp {} /tmp/test-reports/junit/ \;
```
Comment (shakuzen) [inline .circleci/config.yml]: `/tmp/test-reports/junit/` should be `/tmp/test-reports/unit-tests/`

Comment (abesto) [reply .circleci/config.yml]: Same as https://github.com/openzipkin/zipkin-gcp/pull/85#discussion_r200860632, will do the same here as we agree on there.

**Comment by abesto** [comment]
@shakuzen can haz :eyes: one more time?

**Comment by adriancole** [comment]
manually kicked via `curl -X POST https://circleci.com/api/v1.1/project/github/openzipkin/zipkin-aws/tree/release-0.12.1?circle-token=abcdef` from support info here https://discuss.circleci.com/t/force-start-a-branch-build-on-2-0/15517/5

**Comment by adriancole** [comment]
https://circleci.com/gh/openzipkin/zipkin-aws/561 didn't actually run what we think it would have..

@abesto if you have some time, can you help troubleshoot?

**Comment by adriancole** [comment]
I would have expected firstly that the release-0.12.1 tag would have built and run "trigger_publish", but it didn't build and also when I did it manually it didn't do the release stuff

**Comment by abesto** [comment]
Well :hankey:. Sure, I'll look at this next. Expectation management, “next” is something like tomorrow morning.

(I especially like the fact that this probably means none of the CircleCI 2.0 release flows work)

**Pull request merged**